### DISCUSSION TITLE
Distributed instance degraded in case of 1 drive have issues

### DISCUSSION THREAD

**Discussion by AlexZIX** [minio/minio#14175] [Q&A] [answered]
I have distributed minio installation consists of 3 nodes with 4 disks each. Parity set by default EC:4 in this case. When I disconnect up to 4 disks from any node all works fine - users can read and write file without performance issues. But today I got problem with one of the disks. It's performance decreased dramatically but disk still was available in system. When I logged in to this node I see system load 20.0+/ I ran atop and saw that access time to one of the drives are too much (more that 1000 ms.). Other drives works fine. But all my clients got and error with code "SlowDown" and can't download and upload files. I've remove this disk from the system and everything is fine again.

## Expected Behavior
Minio should understand that drive queue is too long and response time is too much and mark this disk as failed.

## Current Behavior
Minio instance with 12 disks stop working even if one disk becomes partially broken (ls on this disk show input/output error).

## Possible Solution
Don't know.

## Steps to Reproduce (for bugs)
1. Eject one disk.
2. Hit with the hammer.
3. Insert back.

## Context
Distributed minio installation stops working.

## Your Environment
* Version used (`minio --version`): 2021-12-10T23:03:39Z
* Server setup and configuration: 3 nodes with 4 drives each in distributed mode.
* Operating System and version (`uname -a`): Linux minio-2.bazis.net 5.4.0-96-generic #109-Ubuntu SMP Wed Jan 12 16:49:16 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

**Comment by klauspost** [comment, marked as answer]
Unmount a partially failed drive:

> For nodes with one or more drives that are either partially failed or operating in a degraded state (increasing disk errors, SMART warnings, timeouts in MinIO logs, etc.), you can safely unmount the drive if the cluster has sufficient remaining healthy drives to maintain read and write quorum. **Missing drives are less disruptive to the deployment than drives that are reliably producing read and write errors.**

https://docs.min.io/minio/baremetal/installation/restore-minio.html

**Reply by AlexZIX** [reply]
It's cool when you have 3 nodes and 12 disks, but in examples in your docs you describe installation on 16 nodes with 200 disks on each. It's 3200 disks. While I will finding the reason of failure my customers goes to competitors, my wife goes to neighbor, my director tells me to go out of office... forever. One failured disk cause entire system unavalability. I even can't login in web console. So only way to recover is to connect to all nodes and check all disks statuses. Are we still discuss about fault-tolerance system? I think that MinIO should do the main disks lifechecks such as load and response time and mark disk as fault with notification. In this case admin can find disk with issue in 5 minutes - not in 5 days.

**Reply by klauspost** [reply]
@AlexZIX Yes, your director *should* fire you if you operate this big a deployment without support in place. If you have 16 nodes with 200 drives each and you have clients, you should get a [support subscription](https://min.io/pricing).  We can help you set up monitoring of your drives and can help you identify your problems in 5 minutes - that is why we offer a "panic" button.

Either way, there is no way to reliably detect partially failed drives when operating on the filesystem. We don't want to take disks offline under the suspicion that it *may* be faulty. In most cases a failing drive is just "really slow". You have to set up appropriate monitoring for your system.
